src/parcel-server.ts

```diff
@@ -14,13 +15,18 @@ export function* createParcelServer(entryPoints: string[], options: ParcelServer
 
   yield watch(bundler, "buildEnd");
 
-  let server = yield bundler.serve(options.port);
-
-  if (process.send) {
-    process.send({ type: "ready" });
-  }
+  let middleware = bundler.middleware();
+  let server = createServer(middleware)
 
   try {
+    yield listen(server, options.port);
+
+    yield receive({ event: "buildEnd" });
```

Comment:
buildEnd is emitted for all builds, including the first one. There is a separate event `bundled` which is only called for the first build, but this seems solid enough for me.

Reply:
This feels so much more robust.